onnx/checker.py

```diff
@@ -68,6 +74,36 @@ def check_tensor_value_info(value_info,
             'TypeProto.value should be either tensor_type or sparse_tensor_type')
 
 
+def check_tensor(tensor):
+    if not isinstance(tensor, TensorProto):
+        raise RuntimeError('You cannot pass an object that is not TensorProto.')
+
+    fields = [field
+              for field in set(mapping.STORAGE_TENSOR_TYPE_TO_FIELD.values())
+              if getattr(tensor, field)]
+    has_raw_field = tensor.HasField('raw_data')
+    if has_raw_field:
+        fields.append('raw_data')
+
+    if len(fields) != 1:
+        raise ValueError(
+            'There should be exactly one data field set: {}'.format(
+                ', '.join(fields)))
+
+    if has_raw_field:
+        if tensor.data_type == TensorProto.STRING:
+            raise ValueError(
+                'STRING data should not be stored in "raw_data" field')
```

Comment:
Aaawkward :) (Awkward in the sense that there is a special case for string here.)